### PR TITLE
fix: Prevent `nix-health` info from printing to `stdout`; add integration test

### LINKED ISSUE #79: stdout is no longer a list of paths built
A regression from #75 

This time, in addition to fixing it we must add a test to prevent this in the future.

One way to fix this: https://github.com/juspay/nix-health/issues/27

### PR DESCRIPTION
**Pull request by shivaraj-bh**
Mark PR ready for review once https://github.com/juspay/nix-health/pull/28 is finalized

resolves #79

### CODE REVIEW

File: tests/integration_test.rs [hunk 1]
```diff
@@ -15,6 +19,26 @@ mod integration_test {
         nixci::logging::setup_logging(true);
     }
 
+    #[tokio::test]
+    /// Run `nixci build` and check if the stdout consists of only /nix/store/* paths
+    async fn nixci_build_output() -> anyhow::Result<()> {
+        let mut cmd = Command::cargo_bin("nixci")?;
```
Comment (srid) [inline tests/integration_test.rs:25]: After this PR, could you open a new PR using this for the rest of the tests in this file (which current looks up `nixci` in $PATH)?

**Review by srid — APPROVED** [review]

File: tests/integration_test.rs [hunk 2]
```diff
@@ -15,6 +19,26 @@ mod integration_test {
...
+        cmd.arg("build")
+            .arg("github:srid/haskell-multi-nix/c85563721c388629fa9e538a1d97274861bc8321");
+
+        let output = cmd.output()?;
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let lines = stdout.lines();
+
+        for line in lines {
+            if !line.starts_with("/nix/store/") {
+                panic!("Unexpected line in stdout: {}", line);
```
Comment (srid) [inline tests/integration_test.rs:35]: Should we be using `panic!` in tests?

**Comment by rsrohitsingh682** [comment]
@shivaraj-bh nixci build fails for this PR.

**Review by rsrohitsingh682 — COMMENTED** [review]
Can you update the nix-health reference? `tracing` branch does not exists now.

**Comment by srid** [comment]
By the way, since this `fix` commit fixes an unreleased changed, I'm going to drop `fix` from commit prefix because I don't want it to appear in the ChangeLog. 

(Is there an appropriate prefix for this?)

**Pull request merged**